### PR TITLE
internal: introduce minicore -- a subset of libcore for testing

### PR DESCRIPTION
**Pull request by matklad**
Clearly, we need one more fixed point iteration loop!

### CODE REVIEW

File: crates/hir_ty/src/tests/coercion.rs [hunk 1]
```diff
@@ -131,60 +122,52 @@ fn infer_let_stmt_coerce() {
 fn infer_custom_coerce_unsized() {
     check_infer(
         r#"
-        struct A<T: ?Sized>(*const T);
-        struct B<T: ?Sized>(*const T);
-        struct C<T: ?Sized> { inner: *const T }
+//- minicore: coerce_unsized
+use core::{marker::Unsize, ops::CoerceUnsized};
```
Comment (matklad) [inline crates/hir_ty/src/tests/coercion.rs:126]: This one now needs a real import!

**Review by flodiebold — APPROVED** [review]
Nice! LGTM.

I should note that I'm also currently cleaning up the hir_ty tests, this one shouldn't conflict too much yet but let me know if you have any other refactoring planned :grimacing:

**Comment by matklad** [comment]
bors r+

No new refactors planned, I'll try to minimize not strictly necessary test changes then!

**Pull request merged**